Gems/Atom/RHI/Vulkan/Code/Source/RHI/SwapChain.cpp

```diff
@@ -135,8 +152,27 @@ namespace AZ
             auto& device = static_cast<Device&>(GetDevice());
             Image* image = static_cast<Image*>(request.m_image);
             RHI::ImageDescriptor imageDesc = request.m_descriptor;
-            imageDesc.m_format = ConvertFormat(m_surfaceFormat.format);
-            RHI::ResultCode result = image->Init(device, m_swapchainNativeImages[request.m_imageIndex], imageDesc);
+            RHI::ResultCode result = RHI::ResultCode::Success;
+
+            // XR swapchains will retrieve the native swapchain image from xr system where as non-xr
+            // swapcahins will create the images internally
+            if (GetDescriptor().m_isXrSwapChain)
+            {
+                RHI::Ptr<XRSwapChainDescriptor> xrSwapChainDescriptor = aznew XRSwapChainDescriptor();
+                xrSwapChainDescriptor->m_inputData.m_swapChainIndex = GetDescriptor().m_xrSwapChainIndex;
+                xrSwapChainDescriptor->m_inputData.m_swapChainImageIndex = request.m_imageIndex;
+                
+                result = GetXRSystem()->GetSwapChainImage(xrSwapChainDescriptor.get());
+                AZ_Assert(result == RHI::ResultCode::Success, "Xr Session creation was not successful");
+
+                result = image->Init(device, xrSwapChainDescriptor->m_outputData.m_nativeImage, imageDesc);
```

Comment:
minor: Since `xrSwapChainDescriptor` pointer is destroyed at the end of the `if` block, it seems unnecessary to reserve heap memory, this can just be:

````
XRSwapChainDescriptor xrSwapChainDescriptor;
xrSwapChainDescriptor.m_inputData.m_swapChainIndex = GetDescriptor().m_xrSwapChainIndex;
xrSwapChainDescriptor.m_inputData.m_swapChainImageIndex = request.m_imageIndex;

result = GetXRSystem()->GetSwapChainImage(&xrSwapChainDescriptor);
AZ_Assert(result == RHI::ResultCode::Success, "Xr Session creation was not successful");

result = image->Init(device, xrSwapChainDescriptor.m_outputData.m_nativeImage, imageDesc);
````

Reply:
There is no difference, with `RHI::Ptr<XRSwapChainDescriptor>` and then passing `xrSwapChainDescriptor .get()` to `GetSwapChainImage` the type is `XRSwapChainDescriptor*`. And using `XRSwapChainDescriptor` and then passing `&xrSwapChainDescriptor ` to `GetSwapChainImage` is the same type `XRSwapChainDescriptor*`.